azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachineEncryptionConfiguration.java

```diff
@@ -92,8 +106,31 @@ public String keyVaultId() {
      * @return URL to the key vault to store the disk encryption key
      */
     public String keyVaultUrl() {
+        if (vaultUri != null) {
+            return vaultUri;
+        }
+
+        String keyVaultDnsSuffix;
+        if (azureEnvironment != null) {
+            keyVaultDnsSuffix = azureEnvironment.keyVaultDnsSuffix();
+
+            if (azureEnvironment.managementEndpoint() != null
+                    && !AzureEnvironment.AZURE.managementEndpoint().equals(azureEnvironment.managementEndpoint())
+                    && AzureEnvironment.AZURE.keyVaultDnsSuffix().equals(azureEnvironment.keyVaultDnsSuffix())) {
```

Comment:
I think it should just `&& isNullOrEmpty(keyVaultDnsSuffix)` and add `AZURE` in the next if.

Reply:
No. Here this "if", I want to know whether the management endpoint is not global, but vault endpoint is. This is the case of `ApplicationTokenCredentials.fromFile`.

All the rest would just go with `azureEnvironment.keyVaultDnsSuffix()`.

Reply:
OK, I will add a NPE protection on `azureEnvironment.keyVaultDnsSuffix()`